src/crimson/os/seastore/random_block_manager/rbm_device.h

```diff
@@ -165,12 +173,28 @@ class RBMDevice : public Device {
   virtual std::string get_device_path() const = 0;
 
   uint64_t get_journal_size() const {
-    return super.journal_size;
+    return shard_info.journal_size;
   }
 
-  static rbm_abs_addr get_journal_start() {
+  static rbm_abs_addr get_journal_start_off() {
     return RBM_SUPERBLOCK_SIZE;
   }
+
+  rbm_abs_addr get_journal_start() {
+    return shard_info.start_offset + get_journal_start_off();
```

Comment:
Right. There is a superblock in RBM. Please see the below.


I'm thinking of a backup super block like other filesystems. This might help to check inconsistency and recovery. So, this part is reserved for now. Also, this can be used as the extension of the existing primary's superblock if the superblock size exceeds the block size. If we want to save the amount of space (super block size x #shard number) by removing a super block in other shards, this is also fine with me for now. But, we need to change the disk layout entirely if those blocks are needed.

Reply:
I'm wondering we still need to adjust the layout in the future.

We need to finally implement dynamic partition across shards, which means the data size for each shard may become dynamic in the future.

In that case, probably starting from `| superblock(s) | circular journals | data |` and static partition the data for each shard can be more generic ?

Reply:
I agree on that we need a further consideration for dynamic partition. I think there are two options as below. But, it seems that I need a study on the dynamic partition to find out pros and cons---there are already many studies, such as [1], that did a research on sharded partitions. 

Option 1 (initial version, but working): | super 1 | journal 1 | data 1 | - | super 2 (reserved) | journal 2 | data 2 |
Option 2: | superblocks(s) | journal(s) | data |

Before doing that, how about merging this PR as is. I'll continue doing this work to find what the better solution is.

[1] Scaling a file system to many cores using an operation log. [SOSP 2017](https://dblp.org/db/conf/sosp/sosp2017.html#BhatECKZ17)